### ISSUE TITLE
Feature Request - Health Checks on DNS Backend Service

### ISSUE THREAD

**Issue by mapl** [TwinProduction/gatus#26]
I would love to see DNS based Health checks

something like

```
type: dns
    url: "udp://127.0.0.1:53"
    queryname: "host.example.org"
    querytype: "A"    
    conditions:
      - "[STATUS] == NOERROR"
      - "[DNS].response == 1.2.3.4" 
```

**Comment by olimpias** [comment]
Hi @TwinProduction , I m planning to work on this issue. However, when I check the structure of the YAML file, `type` property has not been introduced, also saw the previous discussion about it. Do you have any recommendations on how to move forward about the YAML structure for DNS? I think we can embed DNS structure in `services` YAML(Service struct) and add `dns` as a type. If the type is set as DNS, it follows DNS querying, otherwise, default would run. What do you think about it?

**Comment by TwinProduction** [comment]
@olimpias @mapl just to clarify, the idea is to provide a DNS host against which to run a DNS query, correct?

In which case, I think that rather than adding a `type`, we might as well just add a field specifically for DNS queries:
```yaml
services:
  - name: example
    interval: 30s
    url: "udp://8.8.8.8:53" # DNS server to query
    dns:
      query-type: "A"
      query-name: "example.com"
    conditions:
      - "[DNS_RCODE] == NOERROR"
      - "[BODY] == 93.184.216.34"
```

Where `[DNS_RCODE]` is the response code name and `[BODY]` is the relevant query response value (i.e. for a query of type A, that'd be an IP)

| RCODE | NAME |
|:------- |:------- |
| 0 | NOERROR |
| 1 | FORMERR |
| 2 | SERVFAIL |
| 3 | NXDOMAIN
| 4 | NOTIMP |
| 5 | REFUSED |

Even though the `DNS_RCODE` is technically the number and not the name, using the number instead of the name would just be very unfriendly for the users.

Unlike @mapl mentioned, I'd rather not resolve `[STATUS]` at all for services that have a non-nil `dns` field, and use a different placeholder. Even though it could save a bit of code to re-use `[STATUS]`, I'd rather be as explicit as possible. That said, I don't mind re-using the `[BODY]` placeholder.



I'm a dozing off right now, so perhaps I'll have a different opinion today, but let me know what you think 😅

**Comment by olimpias** [comment]
@TwinProduction , yes correct that's the idea in my mind. I totally agree with you. `DNS_RCODE ` should be user friendly. If you don't have any objection or change your opinion about it. I can start working in the evening 🙂.

**Comment by TwinProduction** [comment]
@olimpias Looks good to me 👍

**Comment by mapl** [comment]
Thank you guys for giving this feature request a chance!

> just to clarify, the idea is to provide a DNS host against which to run a DNS query, correct?
Correct, very simple.


I like your DNS Service description. Quite easy to understand

```
conditions:
      - "[DNS_RCODE] == NOERROR"
      - "[BODY] == 93.184.216.34
```

Someone who set this service up, should know what to expect in the [BODY] field.

They're a lot of more DNS RCODEs, and you nailed the first 5, covering 99% of responses.

https://www.iana.org/assignments/dns-parameters/dns-parameters.xhtml#dns-parameters-6

----
_Reference DNS query/response_

```
dig example.com

DiG 9.12.1-P2 <<>> example.com
;; global options: +cmd
;; Got answer:
;; ->>HEADER<<- opcode: QUERY, status: NOERROR, id: 17347
;; flags: qr rd ra ad; QUERY: 1, ANSWER: 1, AUTHORITY: 0, ADDITIONAL: 1

;; OPT PSEUDOSECTION:
; EDNS: version: 0, flags:; udp: 512
;; QUESTION SECTION:
;example.com.                   IN      A

;; ANSWER SECTION:
example.com.            19621   IN      A       93.184.216.34

;; Query time: 7 msec
;; SERVER: 8.8.8.8#53(8.8.8.8)
```

**Comment by TwinProduction** [comment]
Closed by #44 

Released in [v1.7.0](https://github.com/TwinProduction/gatus/releases/tag/v1.7.0)
